cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java

```diff
@@ -0,0 +1,577 @@
+                writeTapChanger(twt, ptcId, twt.getPhaseTapChanger(), CgmesNames.PHASE_TAP_CHANGER, regulatingControlViews, cimNamespace, writer);
+            } else if (twt.hasRatioTapChanger()) {
+                String rtcId = twt.getAliasFromType(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.RATIO_TAP_CHANGER + 1)
+                        .orElseGet(() -> twt.getAliasFromType(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.RATIO_TAP_CHANGER + 2).orElseThrow(PowsyblException::new));
+                writeTapChanger(twt, rtcId, twt.getRatioTapChanger(), CgmesNames.RATIO_TAP_CHANGER, regulatingControlViews, cimNamespace, writer);
+            }
+        }
+
+        for (ThreeWindingsTransformer twt : network.getThreeWindingsTransformers()) {
+            int i = 1;
+            for (ThreeWindingsTransformer.Leg leg : Arrays.asList(twt.getLeg1(), twt.getLeg2(), twt.getLeg3())) {
+                if (leg.hasPhaseTapChanger()) {
+                    String ptcId = twt.getAliasFromType(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.PHASE_TAP_CHANGER + i).orElseThrow(PowsyblException::new);
+                    writeTapChanger(twt, ptcId, leg.getPhaseTapChanger(), CgmesNames.PHASE_TAP_CHANGER, regulatingControlViews, cimNamespace, writer);
+                } else if (leg.hasRatioTapChanger()) {
+                    String rtcId = twt.getAliasFromType(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.RATIO_TAP_CHANGER + i).orElseThrow(PowsyblException::new);
+                    writeTapChanger(twt, rtcId, leg.getRatioTapChanger(), CgmesNames.RATIO_TAP_CHANGER, regulatingControlViews, cimNamespace, writer);
+                }
+                i++;
+            }
+        }
+    }
+
+    private static void writeTapChanger(Identifiable<?> eq, String tcId, TapChanger<?, ?> tc, String defaultType, Map<String, List<RegulatingControlView>> regulatingControlViews, String cimNamespace, XMLStreamWriter writer) throws XMLStreamException {
+        String type = eq.getProperty(cgmesTapChangerReferenceKey(tcId, "type"), defaultType);
+        writeTapChanger(type, tcId, tc, cimNamespace, writer);
+        addRegulatingControlView(tc, tcId, eq, regulatingControlViews);
+        writeHiddenTapChanger(eq, tcId, defaultType, cimNamespace, writer);
+    }
+
+    private static void writeShuntCompensators(Network network, String cimNamespace, Map<String, List<RegulatingControlView>> regulatingControlViews, XMLStreamWriter writer) throws XMLStreamException {
+        for (ShuntCompensator s : network.getShuntCompensators()) {
+            String linearNonlinear;
+            switch (s.getModelType()) {
+                case LINEAR:
+                    linearNonlinear = "Linear";
+                    break;
+                case NON_LINEAR:
+                    linearNonlinear = "Nonlinear";
+                    break;
+                default:
+                    linearNonlinear = "";
+                    break;
```

Comment:
Is this case supposed to happen? If not, throw an AssertionError

Reply:
`AssertionError` thrown